Client/core/CCore.cpp

```diff
@@ -136,12 +136,18 @@ CCore::CCore()
 
     // Create tray icon
     m_pTrayIcon = new CTrayIcon();
+
+    // Create discord rich presence
+    m_pDiscordRichPresence = new CDiscordRichPresence();
```

Comment:
Why not use std::unique_ptr instead of a raw ptr?

Reply:
https://stackoverflow.com/questions/6675651/when-should-i-use-raw-pointers-over-smart-pointers

Reply:
But that is the case of ownership and we should use a smart pointer here. Isn't that right?

Reply:
Now using `shared_ptr`